generation/src/proof_generation/pattern.py

```diff
@@ -415,13 +456,23 @@ def __str__(self) -> str:
 phi1 = MetaVar(1)
 phi2 = MetaVar(2)
 
+"""
+    Class to represent pending evar substitutions
+    It should almost never be used directly, instead use apply_esubst!
+"""
+
 
 @dataclass(frozen=True)
 class ESubst(Pattern):
     pattern: MetaVar | ESubst | SSubst
     var: EVar
     plug: Pattern
```

Comment:
```suggestion

@dataclass(frozen=True)
class ESubst(Pattern):
    """
    Represents evar substitutions over meta-variables.
    It should almost never be used directly, instead use apply_esubst!
    """

    pattern: MetaVar | ESubst | SSubst
    var: EVar
    plug: Pattern
```

Docstrings go inside the thing they document.

Reply:
Ooops, my bad